.github/workflows/sel4bench-pr.yml

```diff
@@ -0,0 +1,107 @@
+# Copyright 2021, Proofcraft Pty Ltd
+#
+# SPDX-License-Identifier: BSD-2-Clause
+
+# Build and run sel4bench on pull requests
+
+name: seL4Bench PR
+
+on:
+  pull_request_target:
+    types: [opened, reopened, synchronize, labeled]
+    paths-ignore:
+      - '**.md'
+
+jobs:
+  code:
+    name: Freeze Code
+    runs-on: ubuntu-latest
+    outputs:
+      xml: ${{ steps.repo.outputs.xml }}
+    steps:
+    - id: repo
+      uses: seL4/ci-actions/repo-checkout@master
+      with:
+        manifest_repo: sel4bench-manifest
+        manifest: master.xml
+
+  build:
+    name: Build
+    needs: code
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        march: [armv7a, armv8a, nehalem, rv64imac]
+    steps:
+    - name: Build
+      uses: seL4/ci-actions/sel4bench@master
+      with:
+        xml: ${{ needs.code.outputs.xml }}
+        march: ${{ matrix.march }}
+    - name: Upload images
+      uses: actions/upload-artifact@v2
+      with:
+        name: images-${{ matrix.march }}
+        path: '*-images.tar.gz'
+
+  hw-run:
+    name: HW Run
+    if: ${{ github.repository_owner == 'seL4' &&
+            (github.event_name == 'pull_request_target' &&
+               github.event.action != 'labeled' &&
+               contains(github.event.pull_request.labels.*.name, 'hw-test') ||
+             github.event_name == 'pull_request_target' &&
```

Comment:
Could merge the `pull_request_target` conditions to
```
            github.event_name == 'pull_request_target' && 
               ((github.event.action != 'labeled' &&
                 contains(github.event.pull_request.labels.*.name, 'hw-test')) ||
                (github.event.action == 'labeled' &&
                 github.event.label.name == 'hw-test')) }}
```

Reply:
I typed a reply for this, but it seems to have disappeared. You're right that this would be shorter, but I prefer to keep the guarding conjunct (`event_name`) together with the event content access (the other `event.*`), so they are easier to move around (those `event.*` fields don't exist for other events so you'd get garbage)